x-pack/plugins/cases/public/containers/user_profiles/api.ts

```diff
@@ -46,7 +46,7 @@ export const bulkGetUserProfiles = async ({
     return [];
   }
 
-  return security.userProfiles.bulkGet({ uids: new Set(uids), dataPath: 'avatar' });
+  return security.userProfiles?.bulkGet?.({ uids: new Set(uids), dataPath: 'avatar' }) ?? [];
```

Comment:
These really shouldn't be necessary but during the tests for the model the security plugin isn't marked as required so the security service is undefined.

For ML attach to cases tests security is defined but `userProfiles` is not.

Reply:
I managed to reproduce it on the feature branch in the ML app. The reason is that the ML app passes the `SecurityPluginSetup` and not the `SecurityPluginStart`. If you see the [contracts](https://github.com/cnasikas/kibana/blob/2dd56cb3fe4362872b79176396dbc51968bc880c/x-pack/plugins/security/public/plugin.tsx#L224) you can see that the `SecurityPluginSetup` does not provide the `userProfiles` services. In this PR https://github.com/elastic/kibana/pull/137830/files#diff-ea33cc971faa8246841cc043dc80397217332cc28808bea25b2016bf2ed2a0f1R140 I fixed it for security solution and oy11. I think it makes sense to have the `SecurityPluginStart` on the UI instead of the `SecurityPluginSetup`.  @legrego what do you think?

cc @darnautov

Reply:
> I think it makes sense to have the SecurityPluginStart on the UI instead of the SecurityPluginSetup

I don't have the full context, so I'm cc'ing @azasypkin to keep me honest. I agree, we should be exposing as much as possible on the `SecurityPluginStart` contract, and reducing the footprint of the `SecurityPluginSetup` contract.

Reply:
thanks for the heads-up @cnasikas, I'll update it on the ML plugin side

Reply:
> I don't have the full context, so I'm cc'ing @azasypkin to keep me honest. I agree, we should be exposing as much as possible on the SecurityPluginStart contract, and reducing the footprint of the SecurityPluginSetup contract.

What @legrego said, even though we can technically expose `userProfiles` via setup contract on the client\browser\UI side, we're not planning to do that to be consistent with the server-side API counterparts as much as possible.